### PR TITLE
Several fixes of shellscript

### PR DESCRIPTION
**Pull request by ghost**
- Replaced as many test utilities with `case` statements as possible;  case statement is ALWAYS built-in while `[` and `test` are NOT.
- Replaced as many `if` statements with `case`; same reason as above.
- Surrounded matching patterns for `case` with parentheses instead of suffixing with right-paren; POSIX lets you do so.
- Also changed coding style completely.

### CODE REVIEW

File: PureLISP.sh [hunk 1]
```diff
@@ -482,38 +467,34 @@ s_apply () {
...
-  fi
+  ;;(*)
+    case "$SREPLREADR" in ("exit")
+      exit 0
+    ;;(\;*)
+      s_repl
+    ;;(*)
+      s_read $SREPLREADR
+      s_eval $SREADR nil
+      s_display $SEVALR && printf $LF
```
Comment (ytaki0801) [inline PureLISP.sh:493]: Conflicted for LF definition.

**Review by ytaki0801 — CHANGES_REQUESTED** [review]
Thank you for pull requests! Unfortunately patch-1 version does not work, so I have reviewed and written comments in patch-1 source code to fix. Please confirm them.

File: PureLISP.sh [hunk 2]
```diff
@@ -22,34 +21,36 @@ cons () {
   eval CAR$CNUM=$1
   eval CDR$CNUM=$2
   CONSR=${CNUM}.conscell
-  CNUM=$((CNUM+1))
+  : $((CNUM++))
```
Comment (ytaki0801) [inline PureLISP.sh:24]: It (and other same changes) does not work with "expecting primarry" message. `$((CNUM=+1)` has no one, but `CNUM`'s value is not being updated...

File: PureLISP.sh [hunk 3]
```diff
@@ -59,41 +60,42 @@ s_strcons () {
...
 
 # S-expression lexical analysis: s_lex
 
 replace_all_posix() {
   set -- "$1" "$2" "$3" "$4" ""
-  until [ _"$2" = _"${2#*"$3"}" ] && eval "$1=\$5\$2"; do
-    set -- "$1" "${2#*"$3"}" "$3" "$4" "$5${2%%"$3"*}$4"
+  while :; do
+    case $2 in (${2#*$3}) break; esac
+    set -- "$5$2" "${2#*"$3"}" "$3" "$4" "$5${2%%"$3"*}$4"
```
Comment (ytaki0801) [inline PureLISP.sh:98]: They do not work, especially $1 assignment without `eval`.

File: PureLISP.sh [hunk 4]
```diff
@@ -108,14 +110,15 @@ s_lex1 () {
   sl1HEAD=${1%% *}
   sl1REST=${1#* }
 
-  if [ ! ${sl1HEAD} = " " ]; then
+  case "$sl1HEAE" in (' ') :
```
Comment (ytaki0801) [inline PureLISP.sh:113]:
```suggestion
  case "$sl1HEAD" in (''|' ') :
```

File: PureLISP.sh [hunk 5]
```diff
@@ -108,14 +110,15 @@ s_lex1 () {
...
     eval "TOKEN$TNUM=\$sl1HEAD"
-    TNUM=$((TNUM+1))
-  fi
+    : $((TNUM++))
+  ;;esac
 
-  if [ ! ${sl1REST} = " " ]; then
-    s_lex1 $sl1REST
-  fi
+  case "$sl1REST" in (*'  ') :
```
Comment (ytaki0801) [inline PureLISP.sh:119]:
```suggestion
  case "$sl1REST" in (' ') :
```

File: PureLISP.sh [hunk 6]
```diff
@@ -108,14 +110,15 @@ s_lex1 () {
...
+  ;;(*) s_lex $sl1REST
```
Comment (ytaki0801) [inline PureLISP.sh:120]:
```suggestion
  ;;(*) s_lex1 $sl1REST
```

File: PureLISP.sh [hunk 7]
```diff
@@ -124,56 +127,56 @@ s_lex () { s_lex0 $1 && s_lex1 $sl0RET; }
...
     SSYNR=$SQUOTER
-  fi
+  ;;esac
 }
 
 s_read () {
   TNUM=0
   s_lex $1
-  SYNPOS=$((TNUM-1))
+  : $((TNUM--))
```
Comment (ytaki0801) [inline PureLISP.sh:179]:
```suggestion
  SYNPOS=$((TNUM-1))
```

File: PureLISP.sh [hunk 8]
```diff
@@ -206,30 +209,30 @@ s_null () { eq $1 nil && SNULLR=$EQR; }
...
+  ;;esac
 }
 
 s_pair () {
   s_null $1
   stackpush $SNULLR
   s_null $2
   stackpop
-  if [ $STACKPOPR = t -o $SNULLR = t ]; then
+  case "$STACKPOPR $SNULLR" in (t *|* t)
```
Comment (ytaki0801) [inline PureLISP.sh:228]:
```suggestion
  case "$STACKPOPR $SNULLR" in (t" "*|*" "t)
```

File: PureLISP.sh [hunk 9]
```diff
@@ -206,30 +209,30 @@ s_null () { eq $1 nil && SNULLR=$EQR; }
...
     SPAIRR=nil
-  else
+  ;;(*)
     atom $1
     stackpush $ATOMR
     atom $2
     stackpop
-    if [ $STACKPOPR = nil -a $ATOMR = nil ]; then
+    case "$STACKPOPR $ATOMR" in (nil nil)
```
Comment (ytaki0801) [inline PureLISP.sh:235]:
```suggestion
    case "$STACKPOPR $ATOMR" in (nil" "nil)
```

**Review by ytaki0801 — CHANGES_REQUESTED** [review]
Sorry, many other comments were not written at the first review so now I added. Again, please confirm them.

File: PureLISP.sh (hunk 2 above)
Comment (ghost) [reply PureLISP.sh:24]: I'm sorry for not haiving understood that increment and decriment are not POSIX-compliant; you are right. As you commented me, `: $((CNUM+=1))` or `CNUM=$((CNUM+1))` are what we were supposed to code. Also thank you for pointing out my mistake.

**Comment by ytaki0801** [comment]
I have merged by hand from GH-TpaeFawzen:patch-1 with some fixes so closed these requests.

**Pull request closed without merging**